datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -1257,51 +1255,6 @@ mod test {
         Ok(())
     }
 
-    #[test]
-    fn test_casting_for_fixed_size_list() -> Result<()> {
```

Comment:
I ported this test to `.slt` as it was the only actual use of datafusion_functions_array in this crate